src/commands/merge.ts

```diff
@@ -114,3 +122,62 @@ export default async function merge({
 
   return ExitCode.Ok;
 }
+
+interface ContextArgs {
+  context: GitHubContext;
+  token: string;
+
+  // for unit testing
+  _mergeFn?: typeof merge;
+}
+export async function mergeByContext({
+  context,
+  token,
+
+  _mergeFn = merge,
+}: ContextArgs): Promise<ExitCode> {
+  let { owner, repo } = readRepository(context);
+  const github = new Octokit({
+    auth: token,
+    userAgent: '@malleatus/nyx automerger (by context)',
+  });
+
+  let event = context.event;
+  if (isStatusEvent(event)) {
```

Comment:
I'm curious why you went with this specific approach. Specifically, why not have a `getPullRequestNumberFromContext` kinda function (moving this logic into that function and leaving this file focused on the logic of doing the merging...

Reply:
That seems like a perfectly good refactor to me

So we'd end up with something like this?
```
export async function mergeByContext({
  context,
  token,

  _mergeFn = merge
}) {
  let prNumber = getPullRequestNumberFromContext(context);
  if(prNumber === undefined) {
    return ExitCode.DidNotRun;
  }

  return _mergeFn({owner, repo, prNumber, token });
}
```